app/assets/javascripts/discourse/tests/integration/components/user-menu/likes-notifications-list-test.js

```diff
@@ -0,0 +1,46 @@
+import { module, test } from "qunit";
+import { setupRenderingTest } from "discourse/tests/helpers/component-test";
+import { render } from "@ember/test-helpers";
+import { cloneJSON } from "discourse-common/lib/object";
+import NotificationFixtures from "discourse/tests/fixtures/notification-fixtures";
+import { hbs } from "ember-cli-htmlbars";
+import pretender from "discourse/tests/helpers/create-pretender";
+
+function getNotificationsData() {
+  return cloneJSON(NotificationFixtures["/notifications"].notifications);
+}
+
+module(
+  "Integration | Component | user-menu | likes-notifications-list",
+  function (hooks) {
+    setupRenderingTest(hooks);
+
+    let notificationsData = getNotificationsData();
+    let queryParams = null;
+    hooks.beforeEach(() => {
+      pretender.get("/notifications", (request) => {
```

Comment:
Can you confirm that we're not leaking state in pretender here? I remember that the helper methods provided in acceptance tests will automatically clean up state in pretender. I don't think we're doing that here.

Reply:
We have a `resetPretender` method that is called after each test:

https://github.com/discourse/discourse/blob/3c60b97bd50dd5b4a0c3de1ff569d4e87f4ee5f3/app/assets/javascripts/discourse/tests/helpers/create-pretender.js#L1141-L1147

Note that this is an integration test so we don't have access to the helper methods that are available to acceptance tests.

Reply:
O icic. We call `resetPretender` on `QUnit.testDone()` so this doesn't leak state at all.